### PR TITLE
Feature/update image upload 40768943

### PR DESCRIPTION
**Pull request by CrazyOptimist**
* update image uploading modal (Figma 7.3 - 7.5)
https://uistorybook.pastel.network/?path=/story/features-add-nft--add-nft
* add commit id to smartcrop.js dep
* fix storybook build error (https://github.com/babel/babel/issues/11622)
* add Tooltip2 common component because Tooltip already exists but not customizable enough
* change optimization slider to use % quality instead of MB because letting user choose size doesn't seem to be implementable

### CODE REVIEW

**Review by ngvtuan — COMMENTED** [review]
Can not close the popup
zoom is not same with the design

**Review by ngvtuan — COMMENTED** [review]
ui issue 1

ui issue 2

ui issue 3

**Review by goal-shevchenko — CHANGES_REQUESTED** [review]
Screen Shot 2021-07-22 at 1 15 27 PM


image

File: src/common/components/Tooltip2/Tooltip2.module.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.arrowDown {
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/common/components/Tooltip2/Tooltip2.module.css:1]: tailwind can't handle different coloring different sides of a border

**Comment by CrazyOptimist** [comment]
I needed to merge PR of image optimization to avoid further conflicts

Removed extra close button on cropping and in full screen

Changed magnification zoom level significantly because with small zoom level it's impossible to see the difference, so now has around 25 times zoom depending on image size

Before/After now has updated design with white background

Removed custom cursor icon from previous design, now it is a standard cursor.

Please ignore other screens, this PR is for "preview" modal with optimization slider.

**Review by ngvtuan — COMMENTED** [review]
This PR is ok to merge. But please resolve the conflict.

**Review by goal-shevchenko — COMMENTED** [review]
Looks good to me then except for merge conflict.

**Review by goal-shevchenko — APPROVED** [review]

File: src/features/nft/addNFT/previewStep/PreviewStepModal.tsx [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+      setOptimizingState('pending')
+      try {
+        const result = await optimizeImage(image, quality, setOptimizedImageURL)
+
+        if (result?.status !== 'cancelled') {
+          setOptimizingState('success')
+        }
+      } catch (error) {
+        setOptimizingState('fail')
+        console.error(error)
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/features/nft/addNFT/previewStep/PreviewStepModal.tsx:85]: And how to log error?

[the remark this answers is not in the document]
Comment (ngvtuan) [reply src/features/nft/addNFT/previewStep/PreviewStepModal.tsx:85]: I think you can use https://www.npmjs.com/package/electron-log if you want to log the error into the production. If this console.log only use for development I think it doesn't need.

**Review by ngvtuan — APPROVED** [review]

**Pull request merged**